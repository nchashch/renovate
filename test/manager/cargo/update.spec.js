const fs = require('fs');
const { updateDependency } = require('../../../lib/manager/cargo/update');

const cargo1toml = fs.readFileSync('test/_fixtures/cargo/Cargo.1.toml', 'utf8');
const cargo5toml = fs.readFileSync('test/_fixtures/cargo/Cargo.5.toml', 'utf8');
const cargo6toml = fs.readFileSync('test/_fixtures/cargo/Cargo.6.toml', 'utf8');
const cargo7toml = fs.readFileSync('test/_fixtures/cargo/Cargo.7.toml', 'utf8');

describe('lib/manager/cargo/update', () => {
  describe('updateDependency()', () => {
    let config;
    beforeEach(() => {
      config = {};
    });
    it('returns same', () => {
      expect(updateDependency('abc', config)).toEqual('abc');
    });
    it('updates normal dependency', () => {
      const upgrade = {
        depName: 'libc',
        lineNumber: 16,
        depType: 'normal',
        newValue: '0.3.0',
      };
      expect(updateDependency(cargo1toml, upgrade)).toEqual(cargo5toml);
    });
    it('updates inline table dependency', () => {
      const upgrade = {
        depName: 'pcap-sys',
        lineNumber: 18,
        depType: 'inlineTable',
        newValue: '0.2.0',
      };
      expect(updateDependency(cargo1toml, upgrade)).toEqual(cargo6toml);
    });
    it('updates standard table table dependency', () => {
      const upgrade = {
        depName: 'winapi',
        lineNumber: 21,
        versionLineNumber: 22,
        depType: 'standardTable',
        newValue: '0.4.0',
      };
      expect(updateDependency(cargo1toml, upgrade)).toEqual(cargo7toml);
    });
    it('does not update in case of error', () => {
      const upgrade = {
        depName: 'libc',
        lineNumber: 13, // Wrong lineNumber
        depType: 'normal',
        newValue: '0.3.0',
      };
      expect(updateDependency(cargo1toml, upgrade)).toEqual(cargo1toml);
    });
  });
});
