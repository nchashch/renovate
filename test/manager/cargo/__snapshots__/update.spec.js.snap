// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/cargo/update updateDependency() handles invalid standard tables gracefully 1`] = `
"[package]
name = \\"standard-tables-example\\"
version = \\"0.1.2\\"

[dependencies.dep1]
optional=true
path=\\"./foo/bar\\"
default-features   = true
version=\\"1.2\\"




[dependencies.dep2]
                    optional=false
         path=\\"./foo/bar\\"
default-features=    true
version    =\\"3.4\\"
[dependencies.dep3]
version=     \\"~12.3.1\\"
default-features=    true
path    =\\"./foo/bar\\"

[dependencies.dep4]
version=     \\"INVALID 1.3.1 VERSION\\"
default-features=    true
path    =\\"./foo/bar\\"

[dependencies.dep5]
vesion = \\"1.2.3\\" # Intentional type
default-features = true
path = \\"./foo/bar\\""
`;

exports[`lib/manager/cargo/update updateDependency() updates inline table dependency 1`] = `
"[package]
name = \\"rustcap\\"
version = \\"0.1.2\\"
description = \\"Wrapper for libpcap\\"
homepage = \\"https://github.com/jmmk/rustcap\\"
repository = \\"https://github.com/jmmk/rustcap\\"
authors = [\\"Michael McLellan <jmikem825@gmail.com>\\"]
keywords = [\\"pcap\\", \\"libpcap\\"]
license = \\"MIT\\"
readme = \\"README.md\\"
documentation = \\"https://docs.rs/rustcap\\"

[workspace]
members = [\\"pcap-sys\\"]

[dependencies]
libc = \\"0.2.43\\"
bitflags = \\"1.0.4\\"
pcap-sys = { version = \\"0.2.0\\", path = \\"pcap-sys\\" }
pnet = { version = \\"0.21.0\\", optional = true, default-features = false}

[target.'cfg(windows)'.dependencies.winapi]
version = \\"0.3.6\\"
features = [\\"ws2def\\", \\"ws2ipdef\\"]

[features]
libpnet = [\\"pnet\\"]

[dev-dependencies]
# libc = \\"0.1.1\\"
# pnet = { version = \\"0.19.0\\", optional = true, default-features = false}
# [target.'cfg(windows)'.dependencies.winapi]
# version = \\"0.3.0\\"
# features = [\\"ws2def\\", \\"ws2ipdef\\"]"
`;

exports[`lib/manager/cargo/update updateDependency() updates normal dependency 1`] = `
"[package]
name = \\"rustcap\\"
version = \\"0.1.2\\"
description = \\"Wrapper for libpcap\\"
homepage = \\"https://github.com/jmmk/rustcap\\"
repository = \\"https://github.com/jmmk/rustcap\\"
authors = [\\"Michael McLellan <jmikem825@gmail.com>\\"]
keywords = [\\"pcap\\", \\"libpcap\\"]
license = \\"MIT\\"
readme = \\"README.md\\"
documentation = \\"https://docs.rs/rustcap\\"

[workspace]
members = [\\"pcap-sys\\"]

[dependencies]
libc = \\"0.3.0\\"
bitflags = \\"1.0.4\\"
pcap-sys = { version = \\"0.1\\", path = \\"pcap-sys\\" }
pnet = { version = \\"0.21.0\\", optional = true, default-features = false}

[target.'cfg(windows)'.dependencies.winapi]
version = \\"0.3.6\\"
features = [\\"ws2def\\", \\"ws2ipdef\\"]

[features]
libpnet = [\\"pnet\\"]

[dev-dependencies]
# libc = \\"0.1.1\\"
# pnet = { version = \\"0.19.0\\", optional = true, default-features = false}
# [target.'cfg(windows)'.dependencies.winapi]
# version = \\"0.3.0\\"
# features = [\\"ws2def\\", \\"ws2ipdef\\"]"
`;

exports[`lib/manager/cargo/update updateDependency() updates standard table table dependency 1`] = `
"[package]
name = \\"rustcap\\"
version = \\"0.1.2\\"
description = \\"Wrapper for libpcap\\"
homepage = \\"https://github.com/jmmk/rustcap\\"
repository = \\"https://github.com/jmmk/rustcap\\"
authors = [\\"Michael McLellan <jmikem825@gmail.com>\\"]
keywords = [\\"pcap\\", \\"libpcap\\"]
license = \\"MIT\\"
readme = \\"README.md\\"
documentation = \\"https://docs.rs/rustcap\\"

[workspace]
members = [\\"pcap-sys\\"]

[dependencies]
libc = \\"0.2.43\\"
bitflags = \\"1.0.4\\"
pcap-sys = { version = \\"0.1\\", path = \\"pcap-sys\\" }
pnet = { version = \\"0.21.0\\", optional = true, default-features = false}

[target.'cfg(windows)'.dependencies.winapi]
version = \\"0.4.0\\"
features = [\\"ws2def\\", \\"ws2ipdef\\"]

[features]
libpnet = [\\"pnet\\"]

[dev-dependencies]
# libc = \\"0.1.1\\"
# pnet = { version = \\"0.19.0\\", optional = true, default-features = false}
# [target.'cfg(windows)'.dependencies.winapi]
# version = \\"0.3.0\\"
# features = [\\"ws2def\\", \\"ws2ipdef\\"]"
`;
